src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py

```diff
@@ -955,7 +955,7 @@ def validate_ssh_key(namespace):
         namespace.ssh_key_value = processed_ssh_key_values
     # if no ssh keys processed, try to generate new key / use existing at root.
     else:
-        namespace.ssh_key_value = _validate_ssh_key_helper("", namespace.generate_ssh_keys)
+        namespace.ssh_key_value = [_validate_ssh_key_helper("", namespace.generate_ssh_keys)]
```

Comment:
@tjprescott, this was the issue, I was the one not properly assigning the result as an array.